bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,348 @@
+import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.READ_PARTIAL_ROW_TIMEOUT_MS;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.google.api.core.InternalApi;
+import com.google.auth.Credentials;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.config.BulkOptions;
+import com.google.cloud.bigtable.config.CallOptionsConfig;
+import com.google.cloud.bigtable.config.CredentialOptions;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.hbase.BigtableExtendedConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.hbase.wrappers.BigtableHBaseSettings;
+import com.google.common.base.Preconditions;
+import io.grpc.Status;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.util.VersionInfo;
+
+/** For internal use only - public for technical reasons. */
+@InternalApi("For internal usage only")
+public class BigtableHBaseClassicSettings extends BigtableHBaseSettings {
+
+  public BigtableHBaseClassicSettings(Configuration configuration) {
+    super(configuration);
+  }
+
+  // <editor-fold desc="Public API">
+  /** For internal use only - public for technical reasons. */
+  @InternalApi("For internal usage only")
+  public BigtableOptions getBigtableOptions() throws IOException {
+    BigtableOptions.Builder bigtableOptionsBuilder = BigtableOptions.builder();
+
+    bigtableOptionsBuilder.setProjectId(getValue(PROJECT_ID_KEY, "Project " + "ID"));
+    bigtableOptionsBuilder.setInstanceId(getValue(INSTANCE_ID_KEY, "Instance ID"));
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+
+    if (appProfileId != null) {
+      bigtableOptionsBuilder.setAppProfileId(appProfileId);
+    }
+
+    String dataHostOverride = configuration.get(BIGTABLE_HOST_KEY);
+    if (dataHostOverride != null) {
+      LOG.debug("API Data endpoint host %s.", dataHostOverride);
+      bigtableOptionsBuilder.setDataHost(dataHostOverride);
+    }
+
+    String adminHostOverride = configuration.get(BigtableOptionsFactory.BIGTABLE_ADMIN_HOST_KEY);
+    if (adminHostOverride != null) {
+      LOG.debug("Admin endpoint host %s.", adminHostOverride);
+      bigtableOptionsBuilder.setAdminHost(adminHostOverride);
+    }
+
+    String portOverrideStr = configuration.get(BigtableOptionsFactory.BIGTABLE_PORT_KEY);
+    if (portOverrideStr != null) {
+      bigtableOptionsBuilder.setPort(Integer.parseInt(portOverrideStr));
+    }
+
+    String usePlaintextStr = configuration.get(BIGTABLE_USE_PLAINTEXT_NEGOTIATION);
+    if (usePlaintextStr != null) {
+      bigtableOptionsBuilder.setUsePlaintextNegotiation(Boolean.parseBoolean(usePlaintextStr));
+    }
+
+    setBulkOptions(bigtableOptionsBuilder);
```

Comment:
The naming implies that this is a setter, which its not.
I think configureBulkOptions() is more appropriate

same for below

Reply:
Instead of `configureBulkOptions()` I went with `buildBulkOptions()` to have similarity between classic & veneer private helper method names. I hope that should be fine.